src/Http/Http.Extensions/src/HttpRequestJsonExtensions.cs

```diff
@@ -31,6 +31,7 @@ public static class HttpRequestJsonExtensions
     /// <param name="cancellationToken">A <see cref="CancellationToken"/> used to cancel the operation.</param>
     /// <returns>The task object representing the asynchronous operation.</returns>
     [RequiresUnreferencedCode(RequiresUnreferencedCodeMessage)]
+    [RequiresDynamicCode(RequiresUnreferencedCodeMessage)]
```

Comment:
This message is specific to "unreferenced" code, not "dynamic" code. Since this is public API, we should have a decent warning message about why this API requires "dynamic" code.